dev-requirements.txt

```diff
@@ -2,37 +2,57 @@
 # These dependencies are not needed to run the Scalyr Agent, but mainly used for
 # running test suites, and mocking Python Objects for the Scalyr Agent.
 
+
 # Testing tools and libraries
 mock==3.0.5
 psutil==5.7.0
-pytest==4.6.9; python_version < '3.0'
-pytest==5.4.3; python_version >= '3.5'
+pytest==5.4.3;
 pytest-coverage
 pytest-timeout==1.3.4
 pytest-benchmark==3.2.3
 pytest-xdist==1.31.0
 coverage==4.5.4
 codecov==2.1.9
 decorator==4.4.1
-PyYAML==5.4.1; python_version >= '3.6'
-PyYAML==5.3.1; python_version == '3.5'
-PyYAML==5.3.1; python_version <= '2.7'
-six==1.13.0
-docker==4.1.0
-# the version of 'requests' library that 'docker' uses as a dependency is higher than we use in agent,
-# to prevent versions conflict, install the appropriate version of 'requests', to force 'docker' to reuse it.
-# NOTE: We can't use requests >= 2.22.0 since we still need to support Python 2.6
-requests==2.15.1
+PyYAML==5.4.1;
+six==1.14.0
 ujson==1.35
-orjson==3.5.2; python_version >= '3.6'
-orjson==2.0.11; python_version == '3.5'
 # Needed by MockHTTPServer class and related tests
 flask==1.1.1
-pathlib2==2.3.5; python_version <= '2.7'
+distro==1.6.0; platform_system != "Windows"
+
 # Used for performance optimized versions of rfc3339_to_* functions
 # NOTE: Not supported on windows
-udatetime==0.0.16
-futures==3.3.0; python_version <= '2.7'
-# Newer versions don't support Python 3.5 anymore
-more-itertools==8.5.0; python_version >= '3.0'
-more-itertools==5.0.0; python_version <= '2.7'
+udatetime==0.0.16; platform_system != "Windows"
+
+# Used in package AMI tests.
+# NOTE: We rely on change in Libcloud which allows us to specify a custom wait_period which makes
+# it less likely for us to hit EC2 rate limits when spinning up and deploying many instances
+# concurrently
+git+https://github.com/apache/libcloud.git@trunk#egg=apache-libcloud; platform_system != "Windows"
+paramiko==2.7.2; platform_system != "Windows"
+
+# benchmarks dependencies
+numpy==1.16.6
+brotli
+
+
+#agent dependencies
+orjson==3.6.4
+# TODO: remove six from the code base.
+six==1.14.0
+requests==2.26.0
+pg8000==1.10.6
```

Comment:
Does this mean we can get rid of third party directory now since those are installed as regular pip dependencies?

Reply:
Correct, I just don't touch anything that can be removed to reduce the amount of noise in the PR.  Everything will be removed at the last step.